### DISCUSSION TITLE
Option to ignore specific renderable (before the target) while doing pick query in filament view

### DISCUSSION THREAD

**Discussion by udayk07** [google/filament#5007] [Q&A]
Expected : 
Have to move a renderable ( in this sample case, black color renderable ) around the given model 

We are trying to implement move a renderable, while user tries to relocate it on touch. In given sample case, a renderable ( black color one ) is attached to the given model. While trying to use pick query, it gives front renderable's position ( as designed in filament ).
Is there any option available to exclude the specific renderable ( front one ), so that pick query should target specific model ?

Thanks in advance

example:

![filament_pick_query](https://user-images.githubusercontent.com/78372917/147853041-7137601a-6a1d-42df-b253-b4c800eca62c.jpg)

**Comment by bejado** [comment]
I don't think there's any easy way to do this at the moment, but @pixelflinger might have some ideas.

**Comment by pixelflinger** [comment]
No there is no way to do this right now. Picking uses the depth buffer (actually it uses the structure buffer which is a half resolution version of the depth buffer, also used for SSAO).

However there is a workaround -- objects that are translucent should not participate to picking, this is because they're not rendered in the SSAO structure buffer. So try to set your black circle as a translucent.
